Lib/ntpath.py

```diff
@@ -311,6 +311,32 @@ def ismount(path):
         return False
 
 
+_reserved_names = frozenset(
+    {'CON', 'PRN', 'AUX', 'NUL', 'CONIN$', 'CONOUT$'} |
+    {f'COM{c}' for c in '123456789\xb9\xb2\xb3'} |
+    {f'LPT{c}' for c in '123456789\xb9\xb2\xb3'}
+)
+def isreserved(path):
+    """Return true if the pathname is reserved by the system."""
+    # Refer to "Naming Files, Paths, and Namespaces":
+    # https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file
+    name = os.fsdecode(basename(path))
+    # '.' and '..' are not reserved.
+    if name == '.' or name == '..':
+        return False
+    # Trailing spaces and dots are reserved.
+    elif name and name[-1] in '. ':
```

Comment:
I suppose the `elif` change is a matter of personal style. I prefer to use `if` with a series of blocks that just return. Anyway...

```suggestion
    elif name.endswith((' ', '.')):
```

Reply:
I think Mr Cannon prefers this style, I may be wrong.